skopt/space/space.py

```diff
@@ -55,13 +55,19 @@ def check_dimension(dimension, transform=None):
         Search space Dimension.
         Each search dimension can be defined either as
 
-        - a `(lower_bound, upper_bound)` tuple (for `Real` or `Integer`
-          dimensions),
-        - a `(lower_bound, upper_bound, "prior")` tuple (for `Real`
-          dimensions),
-        - as a list of categories (for `Categorical` dimensions), or
-        - an instance of a `Dimension` object (`Real`, `Integer` or
-          `Categorical`).
+        - a list or numpy array of choices, for `Categorical` dimension
+        - a 2-, 3- or 4-tuple, for `Real` and `Integer` dimensions, of
+          the form ``(low, high [, prior [, base]])`` (values in square
+          brackets are optional). If both ``low`` and ``high`` are integral
+          numbers (as per the `number.Integral`), a `Integer` dimension is
+          returned, else a `Real` dimension is returned.
+
+          NOTE: For a transitionary period, tuple, list and array currently all
```

Comment:
[`.. note::` in ReST](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html).